src/app.ts

```diff
@@ -1,8 +1,50 @@
-import { Application } from 'pixi.js'
-import Tile from './tile'
+import { Application, interaction, Graphics } from 'pixi.js'
+import Tile, { InteractionContext } from './tile'
+import { toBoard } from './board'
+import { toStartArea } from './start-area'
+import { centerIn, rectsAreIntersecting } from './utils'
 
-const app = new Application(0, 0, { backgroundColor: 0x1099bb })
+export const toApp = (): Application => {
+  return new Application(0, 0, { backgroundColor: 0x1099bb })
+}
 
-app.stage.addChild(new Tile().sprite)
+const positionSpriteInContainersBasedOnCollision = (containers: Graphics[]) => (
+  _: interaction.InteractionEvent,
+  context: InteractionContext
+) => {
+  const { sprite, startPosition } = context
+  const intersectingContainer = containers.find(c => rectsAreIntersecting(c, sprite))
+  if (intersectingContainer) {
+    console.log('found intersecting container')
+    // Drop it in the middle of the container
+    centerIn(intersectingContainer, sprite)
+  } else if (startPosition) {
+    console.log('nope')
+    // Move it bacl
+    sprite.position.set(startPosition.x, startPosition.y)
+  }
+}
 
-export default app
+export const setupApp = (app: Application): void => {
+  // Create elements
+  const startArea = toStartArea()
+  const board = toBoard()
+  const tile = new Tile(app.stage)
+
+  // Add them to stage
+  app.stage.addChild(startArea)
+  app.stage.addChild(board)
+
+  startArea.addChild(tile.sprite)
+
+  /**
+   * Is there maybe a better way to standardise on this without having to
+   * cast these to Graphics?
```

Comment:
Yah, use `DisplayObject` which is the parent class of everything.

Reply:
on it